src/main/scala/dal/UserDal.scala

```diff
@@ -0,0 +1,22 @@
+package dal
+
+import sqlest._
+import sqlest.ast._
+
+import domain.UserID._
+import dal.table._
+import dal.PasswordSaltExtractor._
+
+trait UserDal extends SqlestDb {
+
+
+  def getPasswordSalt(id: UserID): Option[(String,String)]  = {
+
+  	val query =
+      select
+        .from(UserDetailTable)
+        .where(UserDetailTable.userID == id)
+
+    query.extractHeadOption(PasswordSaltExtractor)
```

Comment:
See Slack conversation, we suggested something like this:

      val query =
      select(UserDetailTable.password, UserDetailTable.salt)
        .from(UserDetailTable)
        .where(UserDetailTable.userID == id)

    query.fetchHeadOption

Reply:
Done